types/src/params.rs

```diff
@@ -382,6 +384,10 @@ impl<'a> Id<'a> {
 	}
 }
 
+fn invalid_params(e: impl ToString) -> ErrorObjectOwned {
+	ErrorObject::owned(ErrorCode::InvalidParams.code(), e.to_string(), None::<()>)
```

Comment:
strictly from the jsonrpc spec:

```
code     | message          | meaning
-32602  | Invalid params | Invalid method parameter(s).
```

we are using -32602 => `message == e: impl ToString`

Reply:
but it was like that before this PR as well, it's picking hair I'm fine having a slightly better error message than "Invalid Params" though we could modify it in another and add this extra data in the data field instead of the message.

Reply:
Doesn't feel like a big deal (I'd expect people to match on the code), but having some more detailed thing inthe data field sounds more "correct" (would it be annoying for us to do this?)